assets/css/admin/style.scss

```diff
@@ -99,6 +99,30 @@ body.post-type-download #screen-meta {
 	margin: 0 0 0 20px;
 }
 
+.edd-header-page-title-wrap {
+	margin-top: 7px;
```

Comment:
The margin-top and line-height become unnecessary if we can add `align-items: center;` to the `#edd-header-wrapper` rule (lines 90-94), and then the margin-top should be -7.